### ISSUE TITLE
Why we need line header?, how about removing it for more space?

### ISSUE THREAD

**Issue by t9md** [t9md/atom-narrow#224]
I'm thinking about not showing line/column header by default.

If you are still atom-narrow user, how do you feel removing line/column header?

@lettertwo 
@dan-weaver
@slavaGanzin 
@kompot
@usirin 
@thancock20

The benefit is
- More display area for line content.
- More un-noisiness
- I'm not sure line/column header really gives valuable information.
  - Why we(I/you) need this?
  - If we want to know more **context**, we can preview by placing cursor on item.

I haven't considered this before, because when there are multiple matches within single line, it was impossible to distinguish these items since it highlight was NOT smart.
But after #222 was merged(smart highlight), this confusion would be gone.

### no line , no column(why is this bad?, I think it's NOT bad)

<img width="729" alt="untitled_ ___github_atom-narrow" src="https://user-images.githubusercontent.com/155205/27258252-1cc9835a-5431-11e7-88ac-93cf4660159e.png">

### line only

<img width="759" alt="untitled_ ___github_atom-narrow" src="https://user-images.githubusercontent.com/155205/27258249-f70d18ac-5430-11e7-90a1-354fde8df745.png">

### line and column(current default)

<img width="765" alt="untitled_ ___github_atom-narrow" src="https://user-images.githubusercontent.com/155205/27258254-3e35d12e-5431-11e7-9fb2-7c74e41a54bc.png">

**Comment by thancock20** [comment]
The top picture looks great to me. I really never pay attention to the line numbers in narrow anyway.

**Comment by kompot** [comment]
Agree. Never using these markers line/column markers. Maybe leave it as an option but I never find them useful personally.

**Comment by lettertwo** [comment]
I like the idea of removing the line headers for line/column markers by default, but, i like the concept of line headers in general. For example, I've been hacking around with using them to get display like this in the diagnostics provider:

![image](https://user-images.githubusercontent.com/210630/27261702-74b309a8-5416-11e7-8b6b-8b9a8ddb5a3f.png)

Grammar doesn't recognize the line headers because they aren't in `\d+: \d+: ` form, so no highlighting, but i do think it would be useful to have them for this use case. Another nice-to-have would be to be able to distinguish between 'error' and 'warning' visually, though i also recognize that there might not be much reason for that kind of support beyond this one particular use case.

**Comment by t9md** [comment]
Thanks all. I thought basically many of you would disagree with it , but that was my miss-assumption.

@lettertwo I understand your suggestion, I once thought same thing when I created linter provider.
distinguishing item with color is more tender to brain.

With the text decoration starting to be supported from v1.19-beta. Changing color get really easy.
I removed most of grammar from narrow.
I might remove all grammar rule and achieve all highlight by text-decoration. #223 .

Also you can do your own highlight without depending on narrow-core support.
You can use `ui.onDidRefresh` hook to decorate narrow-editor.
I will show some example later(maybe, no promise).

**Comment by t9md** [comment]
I made it config option, and disabled by default.
So closing now. Thanks for the feedback.

**Comment by t9md** [comment]
released v0.46.0, only installable from Atom-v1.19-beta.

**Comment by t9md** [comment]
Hi, I merged https://github.com/t9md/atom-narrow/pull/248 to differentiate project-header color.
I want project-header standout on multi-project search result.

And I also noticed that I can simplify file and project-header further.
I have 4 style here, how do you think?

#### 1. `# project-name`, `## file-path` style

<img width="864" alt="1" src="https://user-images.githubusercontent.com/155205/31391760-6d29e584-ae12-11e7-8135-686e0a2293db.png">

#### 2. `[project-name]`, `# file-path`(single `#`) style

<img width="864" alt="2" src="https://user-images.githubusercontent.com/155205/31391764-70f75c1e-ae12-11e7-8514-49af10c6d840.png">

#### 3. `[project-name]`, `file-path` style

<img width="880" alt="3" src="https://user-images.githubusercontent.com/155205/31391772-7463e994-ae12-11e7-8589-74fa8438c32d.png">

#### 4. `project-name`, `file-path` style, most minimalistic( I like this ).

<img width="853" alt="4-__-__" src="https://user-images.githubusercontent.com/155205/31391782-7a7dbe0e-ae12-11e7-8b55-2a63b59288c7.png">

**Comment by lettertwo** [comment]
I like option 3; good compromise between minimal and informative.  What would be a benefit of using options 1 or 2, with the markdown-style hashes (`## like this`)? Just a way to see headers without color?

**Comment by t9md** [comment]
@lettertwo Thanks for comment.

Having historical reason, in older version of narrow, file and project header colors are colorized by atom's grammar, using markdown style maker writing grammar rule lot easier.
Now narrow's file and project header is done in atom's decoration feature, I have full control, no risk to match unrelated line, so I no longer need `#` prefix.

**Comment by t9md** [comment]
I made it configurable by simple template.
You can choose  any style,  default is most minimalistic style.
